src/service_daemon.rs

```diff
@@ -740,76 +782,77 @@ impl Zeroconf {
     }
 
     /// Add `addr` in my services that enabled `addr_auto`.
-    fn add_addr_in_my_services(&mut self, addr: Ipv4Addr) {
+    fn add_addr_in_my_services(&mut self, addr: IpAddr) {
         for (_, service_info) in self.my_services.iter_mut() {
             if service_info.is_addr_auto() {
-                service_info.insert_ipv4addr(addr);
+                service_info.insert_ipaddr(addr);
             }
         }
     }
 
     /// Remove `addr` in my services that enabled `addr_auto`.
-    fn del_addr_in_my_services(&mut self, addr: &Ipv4Addr) {
+    fn del_addr_in_my_services(&mut self, addr: &IpAddr) {
         for (_, service_info) in self.my_services.iter_mut() {
             if service_info.is_addr_auto() {
-                service_info.remove_ipv4addr(addr);
+                service_info.remove_ipaddr(addr);
             }
         }
     }
 
     /// Check for IP changes and update intf_socks as needed.
     fn check_ip_changes(&mut self) {
-        // Get the current IPv4 interfaces.
-        let my_ifv4addrs = my_ipv4_interfaces();
+        // Get the current interfaces.
+        let my_ifaddrs = my_ip_interfaces();
 
         // Remove unused sockets in the poller.
         let deleted_addrs = self
             .intf_socks
             .iter()
             .filter_map(|(_, if_sock)| {
-                if !my_ifv4addrs.contains(&if_sock.intf) {
+                if !my_ifaddrs.contains(&if_sock.intf) {
                     if let Err(e) = self.poller.delete(&if_sock.sock) {
```

Comment:
We should also delete the entry in the new `zc.poll_ids`.  Otherwise if the host IP keeps changing, `zc.poll_ids` will grow and never shrink. 

Although deleting an entry in `HashMap` with only the value is not very efficient,  `poll_ids` is small especially if we remove stale entries promptly.

Reply:
Done, indeed better remove stall addresses